frontend/src/ui-components/AcmTable/AcmTable.tsx

```diff
@@ -787,6 +794,33 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
         })
       }
     }
+
+    const findFilterMatch = (filter: string) => advancedFilters.find((filterItem) => filterItem.id === filter)
+    // advanced filtering, won't apply if preprocessed
+    if (!isPreProcessed) {
+      const advancedFilterSelections: FilterSelections = {}
+      activeAdvancedFilters.forEach(({ columnId, value, operator }) => {
+        if (columnId && value && operator) {
+          advancedFilterSelections[columnId] = [value]
+        }
+      })
+      if (advancedFilters.length && Object.keys(advancedFilterSelections).length) {
+        const filterCategories = Object.keys(advancedFilterSelections)
+
+        filteredItems = filteredItems.filter((item: T) => {
+          let isFilterMatch = true
+          filterCategories.every((filter: string) => {
+            const filterItem: ITableAdvancedFilter<T> | undefined = findFilterMatch(filter)
+            /* istanbul ignore next */
+            const isMatch = filterItem?.tableFilterFn(advancedFilterSelections[filter], item) ?? true
+            if (!isMatch) {
+              isFilterMatch = false
+            }
+          })
+          return isFilterMatch
+        })
```

Comment:
@Randy424 This is not quite right. In order for `Array.every()` to short-circuit, you need to return a boolean from the predicate, rather than setting `isFilterMatch` as a side-effect.

Reply:
ack, I confused the callback boundary.